src/sdk/config_fs.js

```diff
@@ -265,104 +440,222 @@ class ConfigFS {
-        const account = await this.get_config_data(account_path, options);
+        let account = await this.get_identity_config_data(account_path, { ...options, silent_if_missing: true });
+        if (!account) {
+            const old_account_path = this._get_old_account_path_by_name(account_name);
+            account = await this.get_identity_config_data(old_account_path, { ...options, silent_if_missing: true });
+        }
+        if (!account && !options.silent_if_missing) {
+            const err = new Error(`Could not find account by name ${account_name}`);
+            err.code = 'ENOENT';
+            throw err;
+        }
         return account;
     }
 
     /**
-     * list_root_accounts returns the root accounts array exists under the config dir
-     * @returns {Promise<Dirent[]>} 
-     */
-    async list_root_accounts(options) {
-        return nb_native().fs.readdir(this.fs_context, this.accounts_dir_path);
+    * @param {fs.Dirent} entry
+    * @param {string} suffix
+    * @returns {boolean}
+    */
+    _is_config_file(entry, suffix) {
+        return entry.name.endsWith(suffix) &&
+            !entry.name.includes(config.NSFS_TEMP_CONF_DIR_NAME) &&
+            !native_fs_utils.isDirectory(entry);
     }
 
     /**
-     * create_account_config_file creates account config file
-     * if account_data.access_keys is an array that contains at least 1 item -
-     * link all item in account_data.access_keys to the relative path of the newly created config file
-     * @param {string} account_name
-     * @param {*} account_data
-     * @param {boolean} symlink_new_access_keys
-     * @param {String[]} old_access_keys
-     * @returns {Promise<void>} 
+     * list_accounts returns the account names array - 
+     * 1. get new accounts names
+     * 2. check old accounts/ dir exists
+     * 3. if old accounts dir exists return the union of new accounts and old accounts names array
+     * 4. else return new account names array
+     * and add accounts from old accounts/ directory if exists
+     * during upgrade - list accounts is a very expensive operation as it's iterating old accounts and adds the entries that still do not appear in the new accounts folder
+     * @returns {Promise<string[]>} 
      */
-    async create_account_config_file(account_name, account_data, symlink_new_access_keys, old_access_keys = []) {
-        const account_path = this.get_account_path_by_name(account_name);
-        await native_fs_utils.create_config_file(this.fs_context, this.accounts_dir_path, account_path, JSON.stringify(account_data));
+    async list_accounts() {
+        const new_entries = await nb_native().fs.readdir(this.fs_context, this.accounts_by_name_dir_path);
+        const new_accounts_names = new_entries
+            .filter(entry => this._is_config_file(entry, SYMLINK_SUFFIX))
+            .map(entry => this._remove_config_file_suffix(entry.name, SYMLINK_SUFFIX));
+
```

Comment:
I can suggest that you do it in 1 iteration instead of 3 (filter, map, convert to set).

I Can share something (in case it helps):
```
        // we want a set with the account names without the temp config file and without name extension
        const account_names_entries_set = entries.reduce((acc, entry) => {
            if (!entry.name.includes(config.NSFS_TEMP_CONF_DIR_NAME)) {
                acc.add(path.parse(entry.name).name);
            }
            return acc;
        }, new Set());
```

Reply:
I did that initially with for.. of but I was asked on the meeting with you and Guy to use filter and map...
If you remember Guy said that running this array is a lot shorter than all those FS ops we do... anyway, i'll try to enhance it a bit but the provided I would like to avoid creating a Set() if I don't need to, and also I need to go over the old accounts as well and add them to the set..